### PR TITLE
Improve IPFS error messages

### PR DESCRIPTION
**Pull request by Ericson2314**

### CODE REVIEW

File: src/libstore/ipfs-binary-cache-store.cc [hunk 1]
```diff
@@ -205,7 +205,14 @@ class IPFSBinaryCacheStore : public Store
 
         if (!optIpnsPath) {
             if (initialIpfsPath != state->ipfsPath)
-                throw Error("The current IPFS address doesn't match the configured one. \n  initial: %s\n  current: %s",
+                throw Error(
+                    "You perfermed store-modfying actions, creating a new store whose IPFS address doesn't match the configured one:\n"
```
Comment (matthewbauer) [inline src/libstore/ipfs-binary-cache-store.cc:209]: performed

Comment (matthewbauer) [reply src/libstore/ipfs-binary-cache-store.cc:209]: store-modifying

**Comment by matthewbauer** [comment]
Do multiline errors display correctly? I would prefer to make the error short (just what went wrong), and include the extra stuff in a `warn()`.

**Comment by Ericson2314** [comment]
@matthewbauer I think it's a bit wordy too, but I want to make things super clear in the short term. We might just make addToStore etc fail in the IPFS case in the long term.
